File: tests/bench/sieve.py
def main():
    N = 5000000
    flags = bytearray(N + 1)
    i = 2
    while i * i <= N:
        if flags[i] == 0:
            j = i * i
            while j <= N:
                flags[j] = 1
                j += i
        i += 1
    count = 0
    i = 2
    while i <= N:
        if flags[i] == 0:
            count += 1
        i += 1
    print(count)
main()

File: tests/bench/sieve.c
#include <stdio.h>
#include <stdint.h>
static uint8_t flags[5000001];
int main(void){
    long N=5000000, count=0;
    for(long i=2;i*i<=N;i++) if(!flags[i]) for(long j=i*i;j<=N;j+=i) flags[j]=1;
    for(long i=2;i<=N;i++) if(!flags[i]) count++;
    printf("%ld\n",count);
    return (int)(count&255);
}
